examples/Consumer/Program.cs

```diff
@@ -132,74 +46,49 @@ public static void Run_Consume(string brokerList, List<string> topics)
                 { "enable.auto.commit", false },
                 { "statistics.interval.ms", 60000 },
                 { "session.timeout.ms", 6000 },
-                { "auto.offset.reset", "smallest" }
+                { "auto.offset.reset", "smallest" },
+                // note: typically, you should treat error_cb events as information only.
+                { "error_cb", (Action<Error>)(error => Console.WriteLine("Error: " + error.Reason)) },
+                { "stats_cb", (Action<string>)(json => Console.WriteLine($"Statistics: {json}")) }
```

Comment:
I think this is ok. Using constants for this (and only this) to try and differentiate feels like trying to cover up a problem to me. My plan for delegate configuration is that you could also pass the name of the delegate as a string (instead of the delegate). in that case, the client could use reflection to find it in any loaded assembly, and automatically use it (not implemented). This makes the config properties much more config like and like the others.

Reply:
delegates as strings sounds like a bad idea, it mixes implementation details with user configuration.

Reply:
What about making ErrorDelegate factory that returns a string tuple,  with a constructor taking the delegate, this gives us type-safety:
```
 { "auto.offset.reset", "tiniest" },
 Confluent.Kafka.OnError((Action<Error>)(error => Console.WriteLine("Error: " + error.Reason)) },
..
```